File: mobile/android/app/src/main/java/io/meshtech/koasta/net/model/Product.kt
package io.meshtech.koasta.net.model

import com.squareup.moshi.JsonClass
import java.math.BigDecimal

@JsonClass(generateAdapter = true)
data class Product (
  val productId: Int,
  val ageRestricted: Boolean,
  val productType: String,
  val productName: String,
  val productDescription: String?,
  val price: BigDecimal,
  val image: String?
)

File: mobile/ios/src/data/models/product.swift
import Foundation

struct Product: Codable {

  let productId: Int
  let ageRestricted: Bool
  let productType: String
  let productName: String
  let productDescription: String?
  let price: Decimal
  let image: String?
}

extension Product {
  var imageUrl: URL? {
    if let image = image, let url = URL(string: image) {
      return url
    }

    return nil
  }
}
